### PR TITLE
PhpUnitMockShortWillReturnFixer - Introduction

### PR DESCRIPTION
**Pull request by kubawerlos**
Replaces https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/3278 - no update from @kingmad since April.

### CODE REVIEW

File: src/Fixer/PhpUnit/PhpUnitShortWillReturnFixer.php [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+            'Risky when PHPUnit classes are overridden or not accessible, or when project has PHPUnit incompatibilities.'
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return $tokens->isAllTokenKindsFound([T_OBJECT_OPERATOR, T_STRING]);
```
Comment (SpacePossum) [inline src/Fixer/PhpUnit/PhpUnitShortWillReturnFixer.php]: `T_CLASS` as well I think and using `PhpUnitTestCaseIndicator::findPhpUnitClasses`? Or do we want to fix these cases everywhere?

Comment (kubawerlos) [reply src/Fixer/PhpUnit/PhpUnitShortWillReturnFixer.php]: Yes, but step-by-step, I'm waiting for Travis and then will push `PhpUnitTestCaseIndicator` with test fix - you can see some tests have invalid PHP syntax. Then I will add more test cases, like uppercases names, `willDoSomething` or `will` being property and fix them as well and then ask for review ;)

**Comment by kubawerlos** [comment]
@SpacePossum done, you can remove `WIP` label and review ;)

Do want to include https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/4160 and rename the fixer?

**Comment by SpacePossum** [comment]
Not sure about one fixer for all mockery-will-fixing... its sounds like a good idea to me to be honest but lets see what others think.

In the meantime, please add a PHP7.3 test case with something like 
`$someMock->method("someMethod")->will($this->returnValue( 10  ,  ));`

File: src/Fixer/PhpUnit/PhpUnitShortWillReturnFixer.php [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\Indicator\PhpUnitTestCaseIndicator;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @author Michał Adamski <michal.adamski@gmail.com>
```
Comment (keradus) [inline src/Fixer/PhpUnit/PhpUnitShortWillReturnFixer.php]: with amount of changes, I believe you can add yourself as well

File: src/Fixer/PhpUnit/PhpUnitShortWillReturnFixer.php [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+        'returnvaluemap' => 'willReturnMap',
+    ];
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'Usage of `->will($this->returnValue(..))` statements must be replaced by its shorter equivalent `->willReturn(...)`.',
```
Comment (keradus) [inline src/Fixer/PhpUnit/PhpUnitShortWillReturnFixer.php]: Pls mention PHPUnit Mock

File: README.rst [hunk 4]
```diff
@@ -1351,6 +1351,13 @@ Choose from the list of available rules:
 
   *Risky rule: this fixer may change functions named ``setUp()`` or ``tearDown()`` outside of PHPUnit tests, when a class is wrongly seen as a PHPUnit test.*
 
+* **php_unit_short_will_return**
```
Comment (keradus) [inline README.rst]: those are not methods of PHPUnit, but PHPUnit Mock. Please reflect that in name.

Comment (kubawerlos) [reply README.rst]: Done

File: src/Fixer/PhpUnit/PhpUnitShortWillReturnFixer.php (hunk 3 above)
Comment (kubawerlos) [reply src/Fixer/PhpUnit/PhpUnitShortWillReturnFixer.php]: Done

**Comment by SpacePossum** [comment]
@kubawerlos can you rebase and squash for me? :)

**Comment by SpacePossum** [comment]
Thank you @kubawerlos.

**Pull request merged**